pkg/aws/aws.go

```diff
@@ -40,36 +41,84 @@ const (
 	messageValidatedPrerequisites = "Validated pre-requisites"
 )
 
+type CloudOption func(*awsCloud)
+
+const (
+	ControlPlaneSecurityGroupIDKey = "controlPlaneSecurityGroupID"
+	WorkerSecurityGroupIDKey       = "workerSecurityGroupID"
+	PublicSubnetListKey            = "PublicSubnetList"
+	VPCIDKey                       = "VPCID"
+)
+
+func WithControlPlaneSecurityGroup(id string) CloudOption {
+	return func(cloud *awsCloud) {
+		cloud.CloudConfig[ControlPlaneSecurityGroupIDKey] = id
+	}
+}
+
+func WithWorkerSecurityGroup(id string) CloudOption {
+	return func(cloud *awsCloud) {
+		cloud.CloudConfig[WorkerSecurityGroupIDKey] = id
+	}
+}
+
+func WithPublicSubnetList(id []string) CloudOption {
+	return func(cloud *awsCloud) {
+		cloud.CloudConfig[PublicSubnetListKey] = id
+	}
+}
+
+func WithVPCName(name string) CloudOption {
+	return func(cloud *awsCloud) {
+		cloud.CloudConfig[VPCIDKey] = name
+	}
+}
+
 type awsCloud struct {
 	client               awsClient.Interface
 	infraID              string
 	region               string
 	nodeSGSuffix         string
 	controlPlaneSGSuffix string
+	CloudConfig          map[string]interface{}
```

Comment:
This doesn't need to be public for consistency.

```suggestion
	cloudConfig          map[string]interface{}
```